### PR TITLE
Add Section on Item State; fixes #507

### LINKED ISSUE #507: Items: Add a chapter about the actual Item state
The Item article currently discusses the Item syntax and usage. So far completely ignored is the question about the Item state, e.g. Initialization, UNDEF, Binding influence etc.

The chapter will probably be short.

Please collect ideas what needs to be mentioned here!

### PR DESCRIPTION
**Pull request by bgilmer77**
Propose to add a section on "Item State" to the documentation on Items to address #507.
Adding an Item state diagram to this contribution would be very useful.
Signed-off-by: Brad Gilmer <brad@gilmer.tv> (github: bgilmer77)

### CODE REVIEW

**Comment by bgilmer77** [comment]
@ThomDietrich here is a starting contribution for Item State.  One thing that I would really like to add is a state diagram for Item, similar to what already exists for Things.  See [here](http://docs.openhab.org/concepts/things.html#status-transitions)

**Comment by bgilmer77** [comment]
Hum.  Obviously some formatting issues - some of the links appear as text when using GitHub 'view'.  Can anyone see what I have done wrong?  Help please.

**Comment by ThomDietrich** [comment]
Hello @bgilmer77, there shouldn't be spaces in a URL, Markown doesn't identify your code as valid links. The links should be:

http://docs.openhab.org/configuration/rules-dsl.html#manipulating-item-states
http://docs.openhab.org/configuration/rules-dsl.html#system-based-triggers

Btw. please do not link the old openHAB 1 wiki any longer. It's still present only for users still using openHAB 1 and should not be a reference for openHAB 2. Either we have an article on the matter or we need to create one

**Comment by bgilmer77** [comment]
@ThomDietrich I believe this initial contribution is complete and ready for review and comment.  Things can be added, but this seems to me to be a reasonable start.

File: configuration/items.md [hunk 1]
```diff
@@ -206,7 +206,46 @@ Number Livingroom_Temperature "Temperature [%.1f °C]"
 The state of an Item depends on the Item type, the Channel bound to the Item, and internal or external events.
 A analogy can be drawn between the state of an Item and the value of a variable in a computer program.
 
-<!-- TODO add a complete description about the actual state of an Item, e.g. Initialization, UNDEF, Binding etc. -->
+{: #item-state}
+#### Item State
+
+This section provides information about what a user can expect regarding the behavior of the state of an Item
```
Comment (ThomDietrich) [inline configuration/items.md]: period?

File: configuration/items.md [hunk 2]
```diff
@@ -206,7 +206,46 @@ Number Livingroom_Temperature "Temperature [%.1f °C]"
...
+
+-  Items are created with a state of `NULL`.
+The Item will remain in this state until it has been acted upon
+-  Subsequent to the creation of an Item, operations in openHAB such as a user interacting with the Item using the `Basic UI`, or a Binding updating the state of an Item will change the state of the Item
+-  Once an Item is being used, its state will depend upon the Thing to which it is linked, or upon the specifics of the Binding that created the Item.
+Example - The state of a Switch may change from `NULL` to `ON` or `OFF`
+-  A Binding may set the state of an Item to `UNDEF` if it looses communications with a Thing (for example, a Z-wave doorbell with a dead battery).
+The Binding may also set the state to `UNDEF` if an error exists in the binding configuration, or under other conditions
+-  An Item may be assigned a particular state as part of the execution of a [Rule]({{base}}/configuration/rules-dsl.html).
+See, particularly, [Manipulating Item States]({{base}}/configuration/rules-dsl.html#manipulating-item-states)
```
Comment (ThomDietrich) [inline configuration/items.md]: This list doesn't follow the style guide defined by https://github.com/openhab/openhab-docs/blob/gh-pages/package.json

File: configuration/items.md [hunk 3]
```diff
@@ -206,7 +206,46 @@ Number Livingroom_Temperature "Temperature [%.1f °C]"
...
 A analogy can be drawn between the state of an Item and the value of a variable in a computer program.
 
-<!-- TODO add a complete description about the actual state of an Item, e.g. Initialization, UNDEF, Binding etc. -->
+{: #item-state}
+#### Item State
+
+This section provides information about what a user can expect regarding the behavior of the state of an Item
+
+-  Items are created with a state of `NULL`.
+The Item will remain in this state until it has been acted upon
```
Comment (ThomDietrich) [inline configuration/items.md]: "acted upon" doesn't add any clarity. What do you want to say with the second sentence?

File: configuration/items.md [hunk 4]
```diff
@@ -206,7 +206,46 @@ Number Livingroom_Temperature "Temperature [%.1f °C]"
...
+-  Subsequent to the creation of an Item, operations in openHAB such as a user interacting with the Item using the `Basic UI`, or a Binding updating the state of an Item will change the state of the Item
```
Comment (ThomDietrich) [inline configuration/items.md]: The first part is redundant and the item would be clearer if it started with "Operations... will change"

File: configuration/items.md [hunk 5]
```diff
@@ -206,7 +206,46 @@ Number Livingroom_Temperature "Temperature [%.1f °C]"
...
+-  Once an Item is being used, its state will depend upon the Thing to which it is linked, or upon the specifics of the Binding that created the Item.
+Example - The state of a Switch may change from `NULL` to `ON` or `OFF`
+-  A Binding may set the state of an Item to `UNDEF` if it looses communications with a Thing (for example, a Z-wave doorbell with a dead battery).
+The Binding may also set the state to `UNDEF` if an error exists in the binding configuration, or under other conditions
```
Comment (ThomDietrich) [inline configuration/items.md]: Feels like this should be the last item in the list

File: configuration/items.md [hunk 6]
```diff
@@ -206,7 +206,46 @@ Number Livingroom_Temperature "Temperature [%.1f °C]"
...
-<!-- TODO add a complete description about the actual state of an Item, e.g. Initialization, UNDEF, Binding etc. -->
+{: #item-state}
+#### Item State
+
+This section provides information about what a user can expect regarding the behavior of the state of an Item
+
+-  Items are created with a state of `NULL`.
+The Item will remain in this state until it has been acted upon
+-  Subsequent to the creation of an Item, operations in openHAB such as a user interacting with the Item using the `Basic UI`, or a Binding updating the state of an Item will change the state of the Item
+-  Once an Item is being used, its state will depend upon the Thing to which it is linked, or upon the specifics of the Binding that created the Item.
```
Comment (ThomDietrich) [inline configuration/items.md]: Isn't that obvious? If you want to emphasize, that the possible states of an Item depend on the Binding, you should make that clearer

File: configuration/items.md [hunk 7]
```diff
@@ -206,7 +206,46 @@ Number Livingroom_Temperature "Temperature [%.1f °C]"
...
+Example - The state of a Switch may change from `NULL` to `ON` or `OFF`
+-  A Binding may set the state of an Item to `UNDEF` if it looses communications with a Thing (for example, a Z-wave doorbell with a dead battery).
+The Binding may also set the state to `UNDEF` if an error exists in the binding configuration, or under other conditions
+-  An Item may be assigned a particular state as part of the execution of a [Rule]({{base}}/configuration/rules-dsl.html).
+See, particularly, [Manipulating Item States]({{base}}/configuration/rules-dsl.html#manipulating-item-states)
+
+
+*N.B.*  Many openHAB users find that it can be very useful to use Persistence and "System started" rules so that their systems behaves in a predictable way after an openHAB restart.
+
+## Restoring States
```
Comment (ThomDietrich) [inline configuration/items.md]: should be h4

File: configuration/items.md (hunk 1 above)
Comment (bgilmer77) [reply configuration/items.md]: Agree.

File: configuration/items.md (hunk 3 above)
Comment (bgilmer77) [reply configuration/items.md]: Second sentence does not add any useful information.  I will delete it.

File: configuration/items.md (hunk 4 above)
Comment (bgilmer77) [reply configuration/items.md]: Agree.  Will delete all before "Operations in openHAB..."

File: configuration/items.md (hunk 6 above)
Comment (bgilmer77) [reply configuration/items.md]: Agree.  Will re-write this sentence to say, "An Item's state may also be set through a Binding which may be reacting to changes in the real world."

File: configuration/items.md (hunk 5 above)
Comment (bgilmer77) [reply configuration/items.md]: Agree.  I will delete the last point about rules.

File: configuration/items.md (hunk 2 above)
Comment (bgilmer77) [reply configuration/items.md]: Okay - I do not have your linter up and running yet.  Let me see if I can get it going and see what it reveals.

File: configuration/items.md (hunk 7 above)
Comment (bgilmer77) [reply configuration/items.md]: Um... what is h4?  I assume you are referring to the coding?

File: configuration/items.md (hunk 2 above)
Comment (bgilmer77) [reply configuration/items.md]: Okay - ran linter and made a new commit fixing formatting errors.  (FWIW, also ran linter against the 'nit' pass and added a new commit to fix that as well.)

**Comment by bgilmer77** [comment]
@ThomDietrich I believe all of your comments are addressed.

File: configuration/items.md [hunk 8]
```diff
@@ -206,7 +206,21 @@ Number Livingroom_Temperature "Temperature [%.1f °C]"
...
 
-<!-- TODO add a complete description about the actual state of an Item, e.g. Initialization, UNDEF, Binding etc. -->
+{: #item-state}
+#### Item State
+
+This section provides information about what a user can expect regarding the behavior of the state of an Item.
+
+-   Items are created with a state of `NULL`.
+
+-   Operations in openHAB such as a user interacting with the Item using the `Basic UI`, or a Binding updating the state of an Item will change the state of the Item
```
Comment (ThomDietrich) [inline configuration/items.md]: backticks should be used for code-related text only. Adding some around "Basic UI" was by mistake?

File: configuration/items.md [hunk 9]
```diff
@@ -206,7 +206,21 @@ Number Livingroom_Temperature "Temperature [%.1f °C]"
...
+
+-   An Item's state may also be set through a Binding which may be reacting to changes in the real world.
+
+-   A Binding may set the state of an Item to `UNDEF` if it looses communications with a Thing (for example, a Z-wave doorbell with a dead battery).
+The Binding may also set the state to `UNDEF` if an error exists in the binding configuration, or under other conditions
+
+*N.B.*  Many openHAB users find that it can be very useful to use Persistence and "System started" rules so that their systems behaves in a predictable way after an openHAB restart.
```
Comment (ThomDietrich) [inline configuration/items.md]: Please add links to both aspects.

File: configuration/items.md [hunk 10]
```diff
@@ -206,7 +206,21 @@ Number Livingroom_Temperature "Temperature [%.1f °C]"
...
+This section provides information about what a user can expect regarding the behavior of the state of an Item.
+
+-   Items are created with a state of `NULL`.
+
+-   Operations in openHAB such as a user interacting with the Item using the `Basic UI`, or a Binding updating the state of an Item will change the state of the Item
+
+-   An Item's state may also be set through a Binding which may be reacting to changes in the real world.
+
+-   A Binding may set the state of an Item to `UNDEF` if it looses communications with a Thing (for example, a Z-wave doorbell with a dead battery).
+The Binding may also set the state to `UNDEF` if an error exists in the binding configuration, or under other conditions
```
Comment (ThomDietrich) [inline configuration/items.md]: Some points are with, some without trailing period...

**Comment by ThomDietrich** [comment]
Hey @bgilmer77, just a few last comments.

Regarding the Restore State article: You've removed it but as it is valuable information we should rather move it to the right spot. Would you please open a separate PR against https://github.com/openhab/openhab-docs/blob/gh-pages/configuration/persistence.md

Thanks!

File: configuration/items.md (hunk 8 above)
Comment (bgilmer77) [reply configuration/items.md]: Agree - will remove backticks

File: configuration/items.md (hunk 9 above)
Comment (bgilmer77) [reply configuration/items.md]: Agree - will add links to Persistence article and System-based-triggers sections.

File: configuration/items.md (hunk 10 above)
Comment (bgilmer77) [reply configuration/items.md]: Will remove all trailing periods in list items.

**Comment by bgilmer77** [comment]
Will open a separate PR against the persistence.md page

**Comment by ThomDietrich** [comment]
Great. Thanks for your contribution. Everything looks good now. Merging! 🎉

**Pull request merged**